transport.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package swim
+
+import (
+	"net"
+	"time"
+)
+
+// Packet is used to provide some metadata about incoming packets from peers
+// over a packet connection
+type Packet struct {
+	// Buffer is the contents of the packet.
+	Buf []byte
+
+	// From has the address of the peer. This is an actual net.Addr so we
+	// can expose some concrete details about incoming packets.
+	Addr net.Addr
```

Comment:
This address is source address (source), right? I think that it needs 2 addresses (src, dst) in packet. Because of gossip protocol, does it need only source address?

Reply:
I already looked hashicorp! But I wonder there is only from address. :)

Reply:
Ah, in this, a packet is not for network side, but for sender side?!